lib/pbench/agent/tool_data_sink.py

```diff
@@ -278,41 +300,205 @@ def terminate(self):
         return 1
 
 
+class BenchmarkRunDir:
+    """BenchmarkRunDir - helper class for handling the benchmark_run_dir
+    directory Redis parameter vs the actual "local" benchmark run directory.
+
+    It is a requirement of the Tool Meister sub-system that the ${pbench_run}
+    directory is always a prefix of the ${benchmark_run_dir}.
+
+    When the pbench CLI starts the Tool Data Sink directly, the local
+    benchmark run directory is the same as the value of the benchmark_run_dir
+    parameter.
+
+    But when the Tool Data Sink runs in a container, the local volume name for
+    the benchmark run directory might be different from the parameter value.
+    Typically, the container is constructed with the default pbench
+    installation of where the ${pbench_run} directory is
+    "/var/lib/pbench-agent".
+
+    The entity responsible for starting the Tool Data Sink container typically
+    mounts a different directory for /var/lib/pbench-agent via 'podman run
+    --volume /srv/data/pbench-run-dir:/var/lib/pbench-agent:Z'.  This leads to
+    a conflict where the external ${pbench_run} path is different the internal
```

Comment:
Not sure there's much benefit in the `${pbench_run}` notation here. And maybe "... path is difference from the mounted path within the container"? (At least, "different from".)